src/sqlfluff/core/rules/base.py

```diff
@@ -1067,11 +1077,45 @@ def eval(self, **kwargs):
                     code, self.name
                 )
             )
-        self._register[code] = dict(code=code, description=description, cls=cls)
+
+        try:
+            assert (
+                "all" in cls.groups
+            ), "Rule {!r} must belong to the 'all' group".format(code)
+            groups = cls.groups
+        except AttributeError:
+
+            raise AttributeError(
```

Comment:
Should we use `raise from` here?

https://www.pythontutorial.net/python-oop/python-raise-from/